api/client/utils.go

```diff
@@ -94,13 +94,14 @@ func (cli *DockerCli) clientRequest(method, path string, in io.Reader, headers m
 	if resp != nil {
 		serverResp.statusCode = resp.StatusCode
 	}
+
 	if err != nil {
-		if strings.Contains(err.Error(), "connection refused") {
+		if strings.Contains(err.Error(), "connection refused") || strings.Contains(err.Error(), "dial unix") || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
This should successfully work around the issue in #15309, but where does the "Get http:///var/run/docker.sock/v1.19/version" message come from in the first place? Whatever code is creating a "HTTP" URL with a local path seems incorrect.